### DISCUSSION TITLE
How to use custom fn+ keybindings with AutoHotkey?

### DISCUSSION THREAD

**Discussion by iseahound** [seerge/g-helper#2853] [Q&A]
GHelper doesn't have any command line arguments, and AutoHotkey can send every key except for the ones that hook onto the fn key. Do you have a way?

I see you can change the last key pressed, but I can't emulate fn with autohotkey.

**Comment by seerge** [comment]
@iseahound hello, you can't emit Fn+ keybindings with external app, as they emit proprietary keycodes on hardware level only :)

**Reply by iseahound** [reply]
Okay, but the advice you give for setting keybinds is to use an external program. But you also say it's not possible to do so here.

**Reply by seerge** [reply]
You can use Autohotkey (or your fingers) to emit this hotkeys https://github.com/seerge/g-helper?tab=readme-ov-file#%EF%B8%8F-keybindings . They are generic and work from any (even external) keyboard

**Reply by iseahound** [reply]
How do I emit `Fn + Ctrl +  F7 / F8 - Flicker-free dimming Down / Up` from autohotkey? This Fn key is a problem

**Reply by seerge** [reply]
@iseahound you can't. But why do you need that ? What are you trying to do ?

**Reply by iseahound** [reply]
Change flicker-free dimming Down / Up using an external keyboard

**Reply by SuperShake87** [reply]
Hello! Thank you very much Seerge! I have a 2024 G14 and gHelper has allowed me to use the gaming laptop like a cold-touch portable daily driver. This is my first post, so thank you again. 

Now, I have the same question as OP.
I have a Nuphy Air75 v2 external keyboard. I am mapping with VIA/QMK. 
I am able to map most keybindings shown here https://github.com/seerge/g-helper?tab=readme-ov-file#%EF%B8%8F-keybindings. 

Here are my problems so far: 
1. Unable to map FN key to any hotkey sequence. Is there an alternative? Such as 

> Fn + F5 / Fn + Shift + F5 - Toggle Performance Modes forwards / backwards
Ctrl + Shift + F5 / Ctrl + Shift + Alt + F5 - Toggle Performance Modes forwards / backwards

2. Unable to map a Visual Mode switcher to external keyboard. Currently, G14 hotkey is Fn+F4 (or just F4 with function lock). Is there a way to map Visual Mode to the external keyboard; such as Ctrl + Shift + F4 or something similar? This is the same question as OP, except regarding Flicker-free dimming assignment/keybind without the FN binding. (EDIT) Is there a way to know exactly what is being emitted by gHelper, so we can map that to the VIA/QMK keyboard?

3: In addition, is there a way to map the FN-LOCK function to the external keyboard? Something like "Ctrl + Shift + C." 

Im not sure if im giving you new information here, but im definitely have the same questions as OP.

**Reply by ByronJim** [reply]
I have the same question, Ghelper has the shortcuts with FN key and when I had a 2019 laptop that has media play shortcuts with FN-key for example FN-F2 prev, FN-F3 play/pause, FN-F4 next and now there are new shortcuts that I dont use so I want to remap these but I dont found a program that can handle FN key only GHelper.

**Comment by IlyaDin** [comment]
**Hi** @seerge,

I think I might have missed out on the keyboard lock function. It's a real pain when I'm working with my iPad sometimes and I forget to lock the keyboard.

**Reply by seerge** [reply]
Hello, I don't think you can disable keyboard completely. Or even if you do - it will be some generic windows/PC way to do that, not specific to asus laptops :)

**Reply by IlyaDin** [reply]
Hello again!

The G-Helper is an essential components of my ASUS laptop.

They work together to configure the display, keyboard, CPU, and GPU. All in one!
I often encounter situations where the keyboard interferes with my work. Instinctively, I began searching for this in G-Helper. Because it is an element that improves the user's experience with a laptop.
